tests/queries/0_stateless/02346_fulltext_index_old_name.sql

```diff
@@ -0,0 +1,35 @@
+DROP TABLE IF EXISTS tab;
+
+-- Index type 'inverted' was renamed to 'full_text' in April 2024.
+-- Such indexes are experimental. Nevertheless test what happens when ClickHouse encounters tables with the old index type.
+
+-- Create a full text index with the old type
+-- This was how it was done in the old days. These days this throws an exception.
+SET allow_experimental_inverted_index = 1;
+CREATE TABLE tab(k UInt64, s String, INDEX idx(s) TYPE inverted(2)) ENGINE = MergeTree() ORDER BY k; -- { serverError ILLEGAL_INDEX };
+
+-- There are unfortunately side effects of this behavior. In particular, if ClickHouse's automatic table load during
+-- startup finds a table with 'inverted'-type indexes created by an older version, it immediately halts as it thinks
+-- the persistence is corrupt. Similarly (but less severely), tables with 'inverted' index cannot be attached.
+-- A backdoor avoids this. Just set allow_experimental_inverted_index = 0 (which is the default).
+SET allow_experimental_inverted_index = 0;
+CREATE TABLE tab(k UInt64, s String, INDEX idx(s) TYPE inverted(2)) ENGINE = MergeTree() ORDER BY k;
```

Comment:
This looks bad in my opinion. It is not documented, but if somebody is sloppy, they can create a table with inverted indices, even though they should use full_text. Cannot we have the backdoor only for attach queries? Then if I am not mistaken we should be able to attach tables with inverted indices, but won't be able to create new ones.

Reply:
@antaljanosbenjamin I fully agree with you but only in a perfect world.

Consider this: Inverted indexes are experimental, enabling the corresponding setting is only fine for testing. Even while testing, anything can - in theory - happen, data loss, corruption, etc. From a product perspective, breaking experimental features is acceptable (it is not nice though).

My expectation is that nobody in their right mind uses inverted indexes in production. The number of affected users should therefore converge towards zero. This PR accommodates for the handful of users where this assumption does not hold. Consider this PR more of a quickfix in a situation where a real fix (i.e. what you propose) isn't worth the trouble as the addressed problems happens very rarely right now and ceases to happen in future. It will cease to happen in future because the backdoor isn't documented and it will be disabled after a transition period at the end of the year.

So in some sense, this PR isn't perfect but it improves over simply breaking old persistences ...

Reply:
> this PR isn't perfect but it improves over simply breaking old persistences

Okay, this is true, so I am okay merging this as is.

Reply:
@rschu1ze I came across this PR and I saw that you already discussed this, but I have one suggestion, not sure have you considered it or not, so what do you think about the following:

- make `allow_experimental_inverted_index` an alias for `allow_experimental_full_text_index` (that way clickhouse-client will report it as a warning, though not a lot of users look into them, but if we are talking about development env, then maybe the situation is different)
- keep throwing exception on SELECT that uses "inverted" indexes (so that users have to migrate to "fulltext")
- allow attaching "inverted" indexes (to make server start)
- forbid creating new tables with "inverted" indexes (to avoid such tables in future)

Since right now it is still possible to create this deprecated index (and only with `allow_experimental_inverted_index=0`) which at least looks odd (even though SELECT is forbidden).

Plus if your workaround will be kept till the end of the year, this time should be enough for all users of this experimental feature to migrate to new index (for whose who do update ClickHouse). And also that way it will be done in the same fashion as for other futures.

Thoughts?

Reply:
@azat I pushed a [follow-up](https://github.com/ClickHouse/ClickHouse/pull/64846) to this yesterday. The situation based on this PR and the follow-up is actually similar to what you describe:
- creating new `inverted` indexes is still possible (assuming `allow_experimental_inverted_index = 1`)
- `inverted` indexes can be attached (again assuming `allow_experimental_inverted_index = 1`)
- to encourage migration: 1. an exception is thrown during usage (making newly created and attached indexes useless) and 2. the docs no longer mention `allow_experimental_inverted_index`.

About warnings shown by clickhouse-client: Yes, developers and CLI people may check. Any GUI or driver that connects to ClickHouse via HTTP will not see the warning and therefore not bother 😢 

I really appreciate your thoughts but in sum, I guess we think too much about gracefully migrating an experimental feature. The number of production use cases is expected to converge to zero. The main reason for pushing this PR and the follow-up is that ClickHouse server refused to start when it encounters old `inverted` indexes (see the PR message) - as long as this blocker is removed I'd be good.

Reply:
>creating new inverted indexes is still possible (assuming allow_experimental_inverted_index = 1)

This still looks odd, what is the reason to allow this?

Reply:
No good reason except that I made this PR in a bit of a hurry and did not find out quickly where to distinguish table "creation" vs. "attaching" in the code. For a non-experimental feature, that's imho okay.